src/Middleware/Spa/SpaProxy/src/SpaProxyLaunchManager.cs

```diff
@@ -0,0 +1,222 @@
+                    CreateNoWindow = false,
+                    UseShellExecute = true,
+                    WindowStyle = ProcessWindowStyle.Normal,
+                    WorkingDirectory = Path.Combine(AppContext.BaseDirectory, _options.WorkingDirectory)
+                };
+                _spaProcess = Process.Start(info);
+            }
+            catch (Exception exception)
+            {
+                _logger.LogError(exception, $"Failed to launch the SPA development server '{_options.LaunchCommand}'.");
+            }
+        }
+
+        public Task StopAsync(CancellationToken cancellationToken)
+        {
+            // We don't need to do anything here since Dispose will take care of cleaning up the process if necessary.
+            return Task.CompletedTask;
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!_disposedValue)
+            {
+                if (disposing)
+                {
+                    // Nothing to do here since ther are no managed resources
+                }
+
+                try
+                {
+                    if (_spaProcess != null && !_spaProcess.HasExited)
+                    {
+                        // Review: Whether or not to do this at all. Turns out that if we try to kill the
+                        // npm.cmd/ps1 process that we start, even with this option we only stop this process
+                        // and the service keeps running.
+                        // Compared to performing Ctrl+C on the window or closing the window for the newly spawned
+                        // process which seems to do the right thing.
+                        // Process.CloseMainWindow seems to do the right thing in this situation and is doable since
+                        // we now start a proxy every time.
+                        // We can't guarantee that we stop/cleanup the proxy on every situation (for example if someone)
+                        // kills this process in a "rude" way, but this gets 95% there.
+                        // For cases where the proxy is left open and where there might not be a "visible" window the recomendation
+                        // is to kill the process manually. (We will not fail, we will simply notify the proxy is "already" up.
+                        if (!_spaProcess.CloseMainWindow())
+                        {
+                            _spaProcess.Kill(entireProcessTree: true);
+                            _spaProcess = null;
+                        }
+                    }
+                }
+                catch (Exception)
+                {
+                    // Avoid throwing if we are running inside the finalizer.
+                    if (disposing)
+                    {
+                        throw;
+                    }
+                }
+
+                _disposedValue = true;
+            }
+        }
+
+        ~SpaProxyLaunchManager()
```

Comment:
Why is there a finalizer?

Reply:
To ensure cleanup of the process in as many situations as possible. This is a development time only package